src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/SqlParser.java

```diff
@@ -275,6 +284,29 @@ private void addOrderByToSelect(Select select, MySqlSelectQueryBlock queryBlock,
         }
     }
 
+    private SQLExpr extractExprFromOrderExpr(SQLSelectOrderByItem sqlSelectOrderByItem) {
+        SQLExpr expr = sqlSelectOrderByItem.getExpr();
+        if (hasNullOrderInBinaryOrderExpr(expr)) {
+                return ((SQLBinaryOpExpr) expr).getLeft();
+            }
+        return expr;
+    }
+
+    private boolean hasNullOrderInBinaryOrderExpr(SQLExpr expr) {
+        if (expr instanceof SQLBinaryOpExpr) {
```

Comment:
This is subjective, but I feel that you could make code cleaner by exiting earlier and splitting conditions, for example (I might be wrong here, but it's should be equivalent code):

```java
        private boolean hasNullOrderInBinaryOrderExpr(SQLExpr expr) {
            if (!(expr instanceof SQLBinaryOpExpr)) {
                return false;
            }

            // check "shape of expression": <identifier> <operator> <NULL>
            SQLBinaryOpExpr binaryExpr = (SQLBinaryOpExpr) expr;
            if (!(binaryExpr.getLeft() instanceof SQLIdentifierExpr) || !(binaryExpr.getRight() instanceof SQLNullExpr) )
                return false;

            // check that operator IS or IS NOT
            SQLBinaryOperator operator = binaryExpr.getOperator();
            return operator == SQLBinaryOperator.Is || operator == SQLBinaryOperator.IsNot;
        }
```

Reply:
Sure, will incorporate this, unless it breaks anything else.

Reply:
Done.